File: OneDrive - Maganti IT Solutions PVT LTD/Desktop/New folder (2)/backend/middleware/error_handler.py
from fastapi import Request
from fastapi.responses import JSONResponse

from config.logger import logger



async def global_exception_handler(
    request: Request,
    exc: Exception
):

    logger.error(
        f"Error on {request.url.path}: {str(exc)}"
    )


    return JSONResponse(

        status_code=500,

        content={

            "success":False,

            "message":"Internal server error"

        }

    )

File: backend/src/middleware/errorHandler.js
const logger = require("../config/logger");


const errorHandler = (

    err,
    req,
    res,
    next

)=>{


    const statusCode =
        err.statusCode || 500;



    // Log complete error internally

    logger.error({

        message: err.message,

        stack: err.stack,

        url: req.originalUrl,

        method: req.method

    });



    const response = {

        success:false,

        message:
            err.message || 
            "Internal Server Error"

    };



    /*
        Hide stack traces in production
    */

    if(process.env.NODE_ENV !== "production"){

        response.stack =
            err.stack;

    }



    res.status(statusCode)
       .json(response);



};


module.exports = errorHandler;